src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1626 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.Pn532.ListPassive;
+using Iot.Device.Pn532.RfConfiguration;
+using Iot.Device.Rfid;
+using IoT.Device.Pn532;
+using System;
+using System.Buffers.Binary;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Spi;
+using System.IO;
+using System.IO.Ports;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Iot.Device.Pn532
+{
+    /// <summary>
+    /// PN532 RFID/NFC reader
+    /// </summary>
+    public class Pn532 : RfidWriteRead, IDisposable
+    {
+        // Timeout 
+        private const int SerialReadTimeOut = 500;
```

Comment:
I feel like this class should take SerialPort or Stream (i.e. coming from SerialPort) as an input and not be polluted with serial port stuff

Reply:
I'll simplify all this and remove it. From page 34 from the documentation, file:///C:/Repos/PN532/docs/141520.pdf, I can get exact timing based on baud rates. I'll use those for the time out.

Reply:
Now, I'll still keep the other properties as it's a different one for SPI and I2C and not very well documented, so din't find a way to make exact ones.